pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -898,6 +898,7 @@ void connectionFailed(PulsarClientException exception) {
         if (System.currentTimeMillis() > createProducerTimeout
                 && producerCreatedFuture.completeExceptionally(exception)) {
             log.info("[{}] Producer creation failed for producer {}", topic, producerId);
+            client.cleanupProducer(this);
```

Comment:
Can we switch the order of the statements to:-
            setState(State.Failed);
            client.cleanupProducer(this);

Reply:
fixed.